requests/__init__.py

```diff
@@ -48,11 +48,20 @@
 __copyright__ = 'Copyright 2016 Kenneth Reitz'
 
 # Attempt to enable urllib3's SNI support, if possible
-try:
-    from .packages.urllib3.contrib import pyopenssl
-    pyopenssl.inject_into_urllib3()
-except ImportError:
-    pass
+# If the user has a recent version of OpenSSL, this
+# isn't needed as Python's core ssl module handles SNI, etc correctly.
+# If the requirements of urllib3.contrib.pyopenssl change,
+# this version should be modified appropraitely.
+#
+# See https://github.com/kennethreitz/requests/issues/3213 as well.
+import ssl
+
+if ssl.OPENSSL_VERSION_INFO < (1, 0, 1):
```

Comment:
Ah, okay I understand. I'm not sure I think a version check is the appropriate mechanism here, although ultimately that'd be the maintainers' call. My reservation about it is scenarios like this:

* User has Python linked against 1.0.1 or 1.0.2 (or even 1.1.0!)
* User has pyOpenSSL and cryptography (linked against OpenSSL 1.1.1 because they got it via a wheel and we linked the latest version)

In the above a version check would not import pyOpenSSL, but pyOpenSSL would actually provide TLS 1.3, which is a desirable feature. An explicit opt-out seems like a better path, although I admit I am not entirely sure how one would accomplish that short of `os.getenv('NO_PYOPENSSL', None)`